src/hooks/useRestyle.ts

```diff
@@ -93,6 +93,9 @@ const useRestyle = <
   ]);
 
   cleanProps.style = calculatedStyle;
+  if (restyleProps.variant) {
```

Comment:
Feels like the util function `filterRestyleProps` above could be a better place to do this